#!/usr/bin/env python

"""

The cfg file defines dynamic parameters, as well as their names, types,
level, description, default values and more. It also determines how the
reconfiguration GUI will look like. The catkin_make command has to be run
for the changes in this file to take effect. This file is usually in the
/cfg subdirectory of a package. More info on:
http://wiki.ros.org/dynamic_reconfigure/Tutorials/HowToWriteYourFirstCfgFile

Parameters can be grouped. Each group has a type that defines how it appears
on the GUI. The types can be either hide, collapse, apply or tab. Here
is an example:

gen = ParameterGenerator()
mygroup = gen.add_group("MyGroup", type="tab")
mygroup.add("ON_OFF", bool_t, 0, "Turns the system on/off", False) ...

                    _________ NOTES _________

Be careful when changing parameters, their attributes are used by other
pieces of code.

"""

PACKAGE = "ema_stim_matrix" # pkg where the server runs

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#       name            type level      description        min,default,max
gen.add("ON_OFF",      bool_t, 1, "Turns the system on/off",        False)
gen.add("Current",     int_t,  2, "Stimulator current",      40,   0,  80)
gen.add("Pulse_Width", int_t,  3, "Stimulator pulse width",   0,   0, 500)
gen.add("Frequency",   int_t,  4, "Stimulator frequency",    48,   0, 300)
gen.add("Channels",    str_t,  5, "Stimulator channels",        "1,2,3,4")

exit(gen.generate(PACKAGE, PACKAGE, "MatrixServer")) # "cfgfilename"
